File: LeetCode/missingNumber.c
#include <stdio.h>

int missingNumber(int n, int arr[n]){
    int missing = n; 
    for(int i =0; i< n; i++){
        missing ^= i ^ arr[i];
    }
    return missing;
}

int main(){
    int n = 6;
    int arr[] = {0,1,2,3,5, 6};
    printf("%d", missingNumber(n, arr));
}

File: LeetCode/missing_number.py
def missing_number(nums):
    arr = []
    for i in range(max(nums) + 1):
        arr.append(0)
    for i in range(len(nums)):
       arr[nums[i]] = 1
    for i in range(len(nums)):
        if arr[i] == 0:
            return i
    return max(nums) + 1
def missingNumber(nums):
    missing = len(nums)  
    for i, num in enumerate(nums):
        missing ^= i ^ num
    
    return missing

print(missing_number([0,1,2,3,4,5,6,7])) 
print(missingNumber([0,1,2]))